### PR TITLE
Fix squadsize achievement improperly triggering

### LINKED ISSUE #586: Infil size upgrades do not prevent "no squad upgrades" achivement
> winning with CI gives you the achievement for never buying squad size upgrades

https://discordapp.com/channels/578606792448409610/639765230569783320/765569354699112448

### PR DESCRIPTION
**Pull request by cannonfodder1**
Closes #586 
After testing this, I am now a retired XCOM 2 achievement hunter

### CODE REVIEW

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc [hunk 1]
```diff
@@ -90,6 +90,7 @@ static function CHEventListenerTemplate CreateStrategyListeners()
 	Template.AddCHEvent('LowSoldiersCovertAction', PreventLowSoldiersCovertActionNag, ELD_OnStateSubmitted, 99);
 	Template.AddCHEvent('OverrideAddChosenTacticalTagsToMission', OverrideAddChosenTacticalTagsToMission, ELD_Immediate, 99);
 	Template.AddCHEvent('PreCompleteStrategyFromTacticalTransfer', PreCompleteStrategyFromTacticalTransfer, ELD_Immediate, 99);
+	Template.AddCHEvent('HasIncreasedSquadSize', HasIncreasedSquadSize, ELD_Immediate, 99);
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc:93]: The hook/event name is too ambiguous - e.g. one could think that it controls the number of slots on the squad select. I suggest `AllowNoSquadSizeUpgradeAchivement`.

Speaking of hooks - I don't see a CHL issue/pr?

**Review by Xymanek — CHANGES_REQUESTED** [review]

**Comment by Xymanek** [comment]
Blocked by discussion in https://github.com/X2CommunityCore/X2WOTCCommunityHighlander/issues/994

**Comment by Xymanek** [comment]
CHL PR merged

**Comment by Xymanek** [comment]
Please rebase (the current master's submodule already points to a CHL version with the event implemented)

**Comment by Xymanek** [comment]
The submodule is still changed in the diff?

**Comment by cannonfodder1** [comment]
> The submodule is still changed in the diff?

Uh oh I might have screwed up, this branch's CHL is up to date with the CHL master, but CI's master's CHL is a few commits behind

**Comment by Xymanek** [comment]
Ideally I would prefer if the submodule completely went away from this PR's diff

**Review by Xymanek — APPROVED** [review]

**Pull request merged**